zarr/tests/test_storage.py

```diff
@@ -1207,7 +1207,7 @@ def test_chunk_nesting(self):
         assert 'foo/10.20.30' in store
         assert b'yyy' == store['foo/10.20.30']
         # N5 reverses axis order
-        # assert b'yyy' == store['foo/30/20/10']
+        assert b'yyy' == store['foo/30/20/10']
```

Comment:
Per https://github.com/zarr-developers/zarr-python/pull/773#pullrequestreview-684161254, I have a heard time understanding how it's possible to support the requirement that multiple keys be supported for the same item. 

This may be part and parcel of the conversation we're having in gitter: https://gitter.im/zarr-developers/community?at=611d562e933ee46b7564044c

Reply:
Update: ah, now I see that they are re-ordered not just `.`/`/` swapped, but I think my same concern holds.

Reply:
I agree with the concern -- it looks like this behavior is expected from the NestedDirectoryStore: https://github.com/zarr-developers/zarr-python/blob/master/zarr/tests/test_storage.py#L1148

Reply:
On my PR I resurrected this. I at least understand know _why_ it's supported, though I think it's a hackdoor of sorts.